### DISCUSSION TITLE
GTX1050  DDA in Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by cwrx777** [RocketCityDynamics/Success-with-Nvidia-545-drivers-in-Ubuntu-22.04-LTS#3] [Q&A] [answered]
Hi,

Thanks for the great article. I have some questions.
1. what is the linux kernel version in your experiment? 
2. have you tried configuring this in a VM?

I have a laptop with a GTX1050 Mobile running Windows Server 2022 with the intention of using [Discrete Device Assignment ](https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/deploy/deploying-graphics-devices-using-dda)of the GPU to the Ubuntu 22.04.1 LTS VM with 6.2.0.-37-generic kernel. 
After adding the [nvidia ppa](https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu), the installed driver is version 545.29.06. 

In the VM, I am able to view the device in `lshw -C Display`. 

I encountered error as shown below
- Failed to allocate NvKmsKapiDevice
- Failed to register device
- Failed to copy vbios to system memory

some other messages:
- can't derive routing for PCI INT A
- PCI INT A: no GST

**Comment by RocketCityDynamics** [comment, marked as answer]
HI @cwrx777. Thanks for chiming in.

I did not try this in a VM, and would only expect it to get harder going down that path.

I did end up swapping out to an AMD 7600 GPU. There were a few issues with theNvidia that I believe were related to the age of my firmware and motherboard. It was one version too old. I could get the environment stable maybe 1-3 times in a row and then it would crash, never to return to a stable state.

I've been enjoying the AMD card. It just works with no questions asked. I saved the Nvidia for a rainy day on a new Intel board, but that's a low priority. My daily driver is now happy. I can use it with FreeCAD, Unity, Godot, and UnrealEngine 5.0 with no issues. OS has been moved to Mint 21. Now that everything is happier, I'll see what I can learn about tuning the AMD for AI. It is currently very happy doing 3D just graphics.
